uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/AbsenceType.php

```diff
@@ -44,3 +44,54 @@ function civicrm_api3_absence_type_delete($params) {
 function civicrm_api3_absence_type_get($params) {
   return _civicrm_api3_basic_get(_civicrm_api3_get_BAO(__FUNCTION__), $params);
 }
+
+
+/**
+ * AbsenceType.calculateToilExpiryDate Specification
+ *
+ * @param array $spec
+ */
+function _civicrm_api3_absence_type_calculatetoilexpirydate_spec(&$spec) {
+  $spec['absence_type_id'] = [
+    'name' => 'absence_type_id',
+    'title' => 'Absence Type ID',
+    'type' => CRM_Utils_Type::T_INT,
+    'api.required' => 1
+  ];
+
+  $spec['date'] = [
+    'name' => 'date',
+    'title' => 'Date to calculate TOIL expiry for',
+    'type' => CRM_Utils_Type::T_DATE,
+    'api.required' => 1
+  ];
+}
+
+/**
+ * AbsenceType.calculateToilExpiryDate API
+ *
+ * @param array $params
+ *   An array of parameters passed to the API
+ *
+ * @return array API result descriptor
+ *
+ * @throws API_Exception
+ */
+function civicrm_api3_absence_type_calculatetoilexpirydate($params) {
+  $absenceType = CRM_HRLeaveAndAbsences_BAO_AbsenceType::findById($params['absence_type_id']);
+
+  if (!$absenceType) {
```

Comment:
The findById() method throws an exception if it can't find a record, so this condition will never match. You can remove it